### ISSUE TITLE
Subsumption error is not reported in certain scenarios involving list-patterns

### ISSUE THREAD

**Issue by alrz** [dotnet/roslyn#66697]
**Version Used**: main

**Steps to Reproduce**:

```cs
class C
{
    void Test(int[] a)
    {
        switch (a)
        {
            case [_] and [.., 0]:
            case [_] and [<0, ..]:
            case [_] and [.., >0]:
            case [_]: // subsumed by previous cases
                break;
        };
    }
}
```

**Expected Behavior**: `The switch case is unreachable. It has already been handled by a previous case or it is impossible to match.`

**Actual Behavior**: No error.
